### PR TITLE
Onion address override

### LINKED ISSUE #283: consider overrides for sites not using onion-location?
On twitter it was pointed out by @alecmuffett that at least two big news orgs: BBC and Buzzfeed have onions but _intentionally_ aren't presenting the onion-location header. In this case we need to decide to:

1. Manually override the scan results using an externally maintained mapping of onions->main site URLs like https://github.com/alecmuffett/real-world-onion-sites
2. Be clearer that we're only looking for sites that will present the "onion available" UI element in Tor Browser. Currently this is explained in the About page but that's it.

### PR DESCRIPTION
**Pull request by chigby**
This pull request slightly reorganizes the scan result data to support overriding the `onion_available` result on a per-site basis. I've done this by:

1. Renaming the `Scan.onion_available` to `Scan.onion_location_header`. This reflects that what the scan is testing for right now is the presence or absence of the header.
2. Add a `onion_address` field to the `Site` model. This is a url field that editors can fill in with the known Onion address for the site.
3. Replace the `Scan.onion_available` field with a property of the same name that checks if the scanned site has an onion address. If so, then we always return `True`, otherwise we return the `onion_location_header` result.

Fixes #283

### CODE REVIEW

File: sites/models.py [hunk 1]
```diff
@@ -32,6 +32,13 @@ class Site(ClusterableModel):
         unique=True,
         help_text='Specify the domain name without the scheme, '
                   'e.g. "example.com" instead of "https://example.com"')
+    onion_address = models.URLField(
+        blank=True,
+        help_text=('The Onion address for this site. This field takes '
+                   'precedence over the Onion-Location header from scan '
+                   'results for determining if a site is available over Onion '
+                   'services.'),
+    )
```
Comment (SaptakS) [inline sites/models.py:41]: I think we will need to add this in the panel for allowing adding/editing onion addresses

**Review by SaptakS — CHANGES_REQUESTED** [review]
Apart from the comment, logically everything else looks good to me.

**Comment by chigby** [comment]
Added a panel for `onion_address`

**Review by SaptakS — APPROVED** [review]
looks good to me.

**Pull request merged**